apps/sms/views/shared/js/settings.js

```diff
@@ -135,6 +135,26 @@ var Settings = {
     }.bind(this));
   },
 
+  // return true if send-read-report switch is ON else false
+  toSendReadReport: function toSendReadReport() {
+    if (!navigator.mozSettings) {
+      return Promise.reject(new Error('The mozSettings API is not available.'));
+    }
+
+    var defer = Utils.Promise.defer();
+    var key = 'message.mms.sendReadReport.enabled';
+    var request = navigator.mozSettings.createLock().get(key);
```

Comment:
I think you can return the `navigator.mozSettings.createLock().get(key)` directly, but we'll refactor this into settings service anyway...